### ISSUE TITLE
Post-processing does not currently allow for using different integer dtypes in classification problems than the reference model

### ISSUE THREAD

**Issue by galv** [mlperf/inference_policies#158]
This was brought up here https://github.com/mlperf/inference/pull/676/files#diff-5ec4194b82d11de403290895677b5c88R41 by @nvzhihanj 

@nvzhihanj proposed allowing the RNN-T model (which is a sequential classification task, so it outputs a single integer ID at each time step) to output int8, int16, int32, or int64, instead of just int64, which is what the reference pytorch model uses. Since there are only ~26 labels in the dataset (basically, the English alphabet with a few misc tokens), which will fit into an int8 (which has 2^8 possible values) just fine, this seems like a harmless deviation from the model.

Therefore, I would propose adding the following sentence here https://github.com/mlperf/inference_policies/blob/master/inference_rules.adoc#71-pre--and-post-processing, to make clear that this sort of change is acceptable in the open division:

- May reduce the precision of classification outputs if the new type still covers all possible outputs (so that outputting int8 instead of the reference model's int64 is allowed when there are 256 or fewer categories in the classification task)

Note that this may not necessarily be post-processing, since it is conceivable (though unlikely) that someone would write a custom softmax or argmax kernel that outputs 8-bit integers directly (as opposed to casting an int64 of int32 value to int8).

I am interested in whether this sort of issue came up for ImageNet, GNMT, etc.

**Comment by tjablin** [comment]
This seems reasonable, but I would rather add it to Section 8.2. The point is that activations can use arbitrary precision. It's really about pre- or post-processing, per se.

**Comment by galv** [comment]
Yes, section 8.2 makes more sense. Thank you.

I assume you meant to write "It's _not_ really about pre- or post-processing, per se."

**Comment by tjablin** [comment]
Yeah. Sorry. Edited.

**Comment by christ1ne** [comment]
May reduce the precision of classification outputs and/or activations .. @tjablin will update a PR in the rules.

**Comment by tjablin** [comment]
https://github.com/mlperf/inference_policies/pull/159

**Comment by galv** [comment]
Thanks @tjablin
